crates/miden-objects/src/transaction/tx_args.rs

```diff
@@ -205,11 +206,16 @@ impl Deserializable for TransactionArgs {
 /// - An executable program defined by a [MastForest] and an associated entrypoint.
 /// - A set of transaction script inputs defined by a map of key-value inputs that are loaded into
 ///   the advice inputs' map such that the transaction script can access them.
+/// - A script arguments key defined as an optional [`Digest`]: if presented, this key will be
+///   automatically placed to the operand stack before the transaction script execution and could be
+///   used to get the script arguments array. See [TransactionScript::with_script_args] for more
+///   info.
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct TransactionScript {
     mast: Arc<MastForest>,
     entrypoint: MastNodeId,
     inputs: BTreeMap<Digest, Vec<Felt>>,
+    script_args_key: Option<Digest>,
```

Comment:
Not for this PR, but I wonder if it makes sense to keep `inputs` (and now `args_key`) inside `TransactionScript`. If a script is meant to be a representation of a program, then we should probably move these into the `TransactionArgs` struct. Let's create an issue to discuss this.

Reply:
Created an issue: https://github.com/0xMiden/miden-base/issues/1417